src/ReleaseNotesUtil/FixerExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#pragma warning disable CA1031 // Do not catch general exception types
+
+namespace ReleaseNotesUtil
+{
+    public static class FixerExtensions
+    {
+        /// <summary>
+        /// Get all the <see cref="CodeFixProvider"/>s that are implemented in the given <see cref="AnalyzerFileReference"/>
+        /// </summary>
+        /// <returns>An array of <see cref="CodeFixProvider"/>s</returns>
+        public static ImmutableArray<CodeFixProvider> GetFixers(this AnalyzerFileReference analyzerFileReference)
+        {
+            if (analyzerFileReference == null)
+            {
+                return ImmutableArray<CodeFixProvider>.Empty;
+            }
+
+            ImmutableArray<CodeFixProvider>.Builder builder = null;
+
+            try
+            {
+                Assembly analyzerAssembly = analyzerFileReference.GetAssembly();
+                IEnumerable<TypeInfo> typeInfos = analyzerAssembly.DefinedTypes;
+
+                foreach (TypeInfo typeInfo in typeInfos)
+                {
+                    if (typeInfo.IsSubclassOf(typeof(CodeFixProvider)))
+                    {
+                        try
+                        {
+                            ExportCodeFixProviderAttribute attribute = typeInfo.GetCustomAttribute<ExportCodeFixProviderAttribute>();
+                            if (attribute != null)
+                            {
+                                builder = builder ?? ImmutableArray.CreateBuilder<CodeFixProvider>();
+                                var fixer = (CodeFixProvider)Activator.CreateInstance(typeInfo.AsType());
+                                if (HasImplementation(fixer))
+                                {
+                                    builder.Add(fixer);
+                                }
+                            }
+                        }
+                        catch
+                        {
+                        }
+                    }
+                }
+            }
+            catch
+            {
+            }
+
+            return builder != null ? builder.ToImmutable() : ImmutableArray<CodeFixProvider>.Empty;
+        }
+
+        /// <summary>
+        /// Check the method body of the Initialize method of an analyzer and if that's empty,
+        /// then the analyzer hasn't been implemented yet.
+        /// </summary>
+        private static bool HasImplementation(CodeFixProvider fixer)
```

Comment:
Wow interesting approach :). Do we have fixers that register FixableDiagnosticIds and still have empty implementation? I would just change those fixers to return no fixable diagnostic IDs, in which case `HasImplementation == !FixableDiagnosticIds.IsEmpty`

Reply:
Or even consider deleting those fixers... however, given you are executing the tool even for prior releases, you probably need to retain this logic..

Reply:
I just copied this from GenerateAnalyzerRulesets, so ummm...I don't know.  🙂

Reply:
Oh well, I don't even recall writing this now :)